zcash_client_sqlite/src/wallet.rs

```diff
@@ -588,6 +686,70 @@ pub fn get_nullifiers<P>(
     Ok(res)
 }
 
+pub fn get_all_nullifiers<P>(
+    wdb: &WalletDb<P>,
+) -> Result<Vec<(AccountId, Nullifier)>, SqliteClientError> {
+    // Get the nullifiers for the notes we are tracking
+    let mut stmt_fetch_nullifiers = wdb.conn.prepare(
+        "SELECT rn.id_note, rn.account, rn.nf
+            FROM received_notes rn",
+    )?;
+    let nullifiers = stmt_fetch_nullifiers.query_map(NO_PARAMS, |row| {
+        let account = AccountId(row.get(1)?);
+        let nf_bytes: Vec<u8> = row.get(2)?;
+        Ok((account, Nullifier::from_slice(&nf_bytes).unwrap()))
+    })?;
+
+    let res: Vec<_> = nullifiers.collect::<Result<_, _>>()?;
+    Ok(res)
+}
+
+#[cfg(feature = "transparent-inputs")]
+pub fn get_unspent_transparent_outputs<P: consensus::Parameters>(
+    wdb: &WalletDb<P>,
+    address: &TransparentAddress,
+    max_height: BlockHeight,
```

Comment:
`max_height` is ambiguous; it could be the height up to which the utxo can have been created, or the height up to which we will look to see whether it was spent.